nest-city-account/src/oauth2/oauth2.module.ts

```diff
@@ -0,0 +1,38 @@
+import { MiddlewareConsumer, Module, NestModule } from '@nestjs/common'
+
+import { OAuth2Controller } from './oauth2.controller'
+import { OAuth2Service } from './oauth2.service'
+import { AuthorizationRequestGuard } from './guards/authorization-request.guard'
+import { TokenRequestGuard } from './guards/token-request.guard'
+import { AuthorizationPayloadGuard } from './guards/authorization-payload.guard'
+import { OAuth2ValidationSubservice } from './subservices/oauth2-validation.subservice'
+import ThrowerErrorGuard from '../utils/guards/errors.guard'
+import { CognitoSubservice } from '../utils/subservices/cognito.subservice'
+import { OAuth2ExceptionFilter } from '../utils/filters/oauth2.filter'
+import { PrismaService } from '../prisma/prisma.service'
+import { NoCacheMiddleware } from '../utils/middlewares/no-cache.middleware'
+import { TokenRequestValidationPipe } from './pipes/token-request-validation.pipe'
+
+@Module({
+  imports: [],
+  providers: [
+    OAuth2Service,
+    PrismaService,
+    CognitoSubservice,
+    ThrowerErrorGuard,
+    OAuth2ValidationSubservice,
+    AuthorizationRequestGuard,
+    TokenRequestGuard,
+    AuthorizationPayloadGuard,
+    OAuth2ExceptionFilter,
+    TokenRequestValidationPipe,
+  ],
+  exports: [AuthorizationRequestGuard, TokenRequestGuard],
+  controllers: [OAuth2Controller],
+})
+export class OAuth2Module implements NestModule {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  configure(consumer: MiddlewareConsumer): any {
```

Comment:
```suggestion
  configure(consumer: MiddlewareConsumer) {
```
no need to specify type here